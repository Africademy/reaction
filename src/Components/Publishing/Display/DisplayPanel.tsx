import { get, once } from "lodash"
import React from "react"
import styled, { StyledFunction } from "styled-components"
import Colors from "../../../Assets/Colors"
import { crop, resize } from "../../../Utils/resizer"
import { track } from "../../../Utils/track"
import { Fonts } from "../Fonts"
import { VideoControls } from '../Sections/VideoControls'

interface DisplayPanelProps extends React.HTMLProps<HTMLDivElement> {
  campaign: any
  isMobile?: boolean
  unit: any
}

interface DivUrlProps extends React.HTMLProps<HTMLDivElement> {
  coverUrl?: string
  hoverImageUrl?: string
  isMobile?: boolean
  imageUrl?: string
  onMouseEnter: any
  onMouseLeave: any
}

@track()
export class DisplayPanel extends React.Component<DisplayPanelProps, any> {
  private video: HTMLVideoElement

  constructor(props) {
    super(props)
    this.openLink = this.openLink.bind(this)

    this.state = {
      isPlaying: false
    }
  }

  @track(once(props => ({
    action: "Impression",
    entity_type: "display_ad",
    campaign_name: props.campaign.name,
    unit_layout: "panel"
  })))
  componentDidMount() {
    this.video.onended = () => {
      this.setState({ isPlaying: false })
    }
  }

  // Rather than using an <a /> tag, which bad markup can oftentimes break during
  // SSR, use JS to open link.
  @track((props, [e]) => ({
    action: "Click",
    label: "Display ad clickthrough",
    entity_type: "display_ad",
    campaign_name: props.campaign.name,
    unit_layout: "panel"
  }))
  openLink(e) {
    e.preventDefault()
    window.open(this.props.unit.link.url, '_blank')
  }

  onMouseLeave = () => {
    this.video.pause()
  }

  playVideo = () => {
    if (this.video) {
      if (this.video.paused) {
        this.video.play()
        this.setState({ isPlaying: true })
      } else {
        this.video.pause()
        this.setState({ isPlaying: false })
      }
    }
  }

  renderVideo = (url) => {
    const { isMobile } = this.props
    return (
      <VideoContainer onClick={isMobile && this.playVideo}>
        {!this.state.isPlaying &&
          <VideoCover>
            {isMobile &&
              <VideoControls mini />
            }
          </VideoCover>
        }
        <video src={url} controls={false} playsInline ref={video => (this.video = video)} />
      </VideoContainer>
    )
  }


  render() {
    const { unit, campaign, isMobile } = this.props
    const url = get(unit.assets, '0.url', '')
    const cover = unit.cover_image_url || ''
    const imageUrl = crop(url, { width: 680, height: 284 })
    const hoverImageUrl = resize(unit.logo, { width: 680 })
    const coverUrl = crop(cover, { width: 680, height: 284 })
    const isVideo = url.includes("mp4")

    return (
      <Wrapper onClick={!isMobile && this.openLink}>
        <DisplayPanelContainer
          onMouseEnter={!isMobile && this.playVideo}
          onMouseLeave={!isMobile && this.onMouseLeave}
          imageUrl={imageUrl}
          isMobile={isMobile}
          hoverImageUrl={hoverImageUrl}
          coverUrl={coverUrl}>

          {isVideo
            ? this.renderVideo(url)
            : <Image />
          }
          <div onClick={isMobile && this.openLink}>
            <Headline>
              {unit.headline}
            </Headline>

            <Body dangerouslySetInnerHTML={{
              __html: unit.body
            }} />

            <SponsoredBy>
              {`Sponsored by ${campaign.name}`}
            </SponsoredBy>
          </div>
        </DisplayPanelContainer>
      </Wrapper>
    )
  }
}

const Wrapper = styled.div`
  cursor: pointer;
  text-decoration: none;
  color: black;
  margin-top: 50px;
`

const Image = styled.div`
  margin-bottom: 15px;
  width: 100%;
  height: 142px;
  background-color: black;
  box-sizing: border-box;
`

const VideoCover = Image.extend`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
`

const Div: StyledFunction<DivUrlProps> = styled.div

const DisplayPanelContainer = Div`
  display: flex;
  flex-direction: column;
  border: 1px solid ${Colors.grayRegular};
  padding: 20px;
  max-width: 360px;
  box-sizing: border-box;
  ${Image} {
    background: url(${props => (props.imageUrl ? props.imageUrl : "")}) no-repeat center center;
    background-size: cover;
  }
  ${VideoCover} {
    background: url(${props => (props.coverUrl ? props.coverUrl : "")}) no-repeat center center;
    background-size: cover;
  }
  &:hover {
    ${Image} {
      ${props =>
    props.hoverImageUrl
      ? `
          background: black url(${props.hoverImageUrl}) no-repeat center center;
          background-size: contain;
          border: 10px solid black;
        `
      : ""}
    }
    ${VideoCover} {
      ${props => !props.isMobile && "display: none;"}
    }
  }
`

const Headline = styled.div`
  ${Fonts.unica("s16", "medium")} line-height: 1.23em;
  margin-bottom: 3px;
`

const Body = styled.div`
  ${Fonts.garamond("s17")} line-height: 1.53em;
  margin-bottom: 20px;
  a {
    color: black;
  }
`

const SponsoredBy = styled.div`
  ${Fonts.avantgarde("s11")} color: ${Colors.grayRegular};
`

const VideoContainer = Image.extend`
  position: relative;
  overflow: hidden;
  display: flex;
  align-items: center;
  video {
    object-fit: cover;
    object-position: 50%;
    width: 100%;
  }
`
