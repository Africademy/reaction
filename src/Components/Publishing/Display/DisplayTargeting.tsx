import { data as sd } from "sharify"

export const targetingData = (id: string, pageType: string) => {
  const isTesting = sd.AD_TESTING

  return {
    is_testing: isTesting,
    page_type: pageType,
    post_id: id,
  }
}

export const is300x50AdUnit = (unit: string): boolean => {
  return unit.slice(-3) === "x50"
}
