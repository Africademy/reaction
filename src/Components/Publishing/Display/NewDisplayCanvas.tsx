import { color, Flex } from "@artsy/palette"
import { AdDimension, AdUnit } from "Components/Publishing/Typings"
import React, { SFC } from "react"
import { Bling as GPT } from "react-gpt"
import styled from "styled-components"

interface DisplayCanvasProps {
  adUnit?: AdUnit
  adDimension?: AdDimension
  displayNewAds?: boolean
}

export const NewDisplayCanvas: SFC<DisplayCanvasProps> = props => {
  const { adUnit, adDimension, displayNewAds } = props

  if (!displayNewAds) {
    return null
  }

  const [width, height] = adDimension.split("x").map((a: string) => parseInt(a))

  return (
    <DisplayCanvasContainer flexDirection="column" m="auto" p={2}>
      <GPT adUnitPath={`/21805539690/${adUnit}`} slotSize={[width, height]} />
    </DisplayCanvasContainer>
  )
}

const DisplayCanvasContainer = styled(Flex)`
  min-height: fit-content;
  box-sizing: border-box;
  background: ${color("black5")};
  text-align: center;
`

// Set names for tests and DOM
DisplayCanvasContainer.displayName = "DisplayCanvasContainer"
