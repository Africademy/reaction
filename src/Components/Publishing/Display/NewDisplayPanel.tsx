import { Box, color, Flex } from "@artsy/palette"
import { AdDimension, AdUnit } from "Components/Publishing/Typings"
import React, { SFC } from "react"
import { Bling as GPT } from "react-gpt"
import styled from "styled-components"

export interface DisplayPanelProps extends React.HTMLProps<HTMLDivElement> {
  adUnit?: AdUnit
  adDimension?: AdDimension
}

export const NewDisplayPanel: SFC<DisplayPanelProps> = props => {
  const { adDimension, adUnit } = props
  const [width, height] = adDimension.split("x").map((a: string) => parseInt(a))

  return (
    <Wrapper color="black100">
      <DisplayPanelContainer
        flexDirection="column"
        className="DisplayPanel__DisplayPanelContainer"
        p="40px auto"
        m="auto"
        width={width}
        height={height}
      >
        <GPT
          adUnitPath={`/21805539690/${adUnit}`}
          targeting={{
            is_testing: "no",
            page_type: "article",
            post_id: "5cdd7407cb1f5f00189c8205",
          }}
          slotSize={[width, height]}
        />
      </DisplayPanelContainer>
    </Wrapper>
  )
}

const Wrapper = styled(Box)`
  cursor: pointer;
  text-decoration: none;
  max-width: 360px;
`
const DisplayPanelContainer = styled(Flex)`
  max-width: 360px;
  box-sizing: border-box;
  background: ${color("black5")};
  text-align: center;
`

// Set names for tests and DOM
DisplayPanelContainer.displayName = "DisplayPanelContainer"
Wrapper.displayName = "Wrapper"
