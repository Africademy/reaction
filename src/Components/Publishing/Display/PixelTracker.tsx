import React from "react"
import styled from "styled-components"

export const PixelTracker = props => {
  let url = props.pixel_tracking_code
  if (!url) {
    return null
  }

  const hasTimestamp = ["serving-sys", "doubleclick"].some(str =>
    url.includes(str)
  )
  if (hasTimestamp) {
    url = url.replace(
      "[timestamp]",
      String(new Date().getMilliseconds() / 1000)
    )
  }

  return <TrackerImage width={1} height={1} src={url} />
}

const TrackerImage = styled.img`
  display: none;
`
