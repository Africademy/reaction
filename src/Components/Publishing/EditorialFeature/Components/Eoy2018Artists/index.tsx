import { color, Serif } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import { ArticleProps } from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import { ImageSetPreview } from "Components/Publishing/Sections/ImageSetPreview"
import React from "react"
import styled from "styled-components"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export const BORDER_WIDTH = 4

export class Eoy2018Artists extends React.Component<ArticleProps> {
  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionIsHeaderImage = (section, i) => {
    const {
      article: { sections },
    } = this.props
    const isImage = section.type === "image_collection"
    const sectionBefore = i !== 0 && sections[i - 1]
    const sectionBeforeIsHeader = this.sectionIsHeader(sectionBefore)

    return isImage && sectionBeforeIsHeader
  }

  sectionHeader = (section, i) => {
    const {
      article: { sections },
    } = this.props
    const sectionAfter = sections[i + 1]
    const sectionAfterisImage = this.sectionIsHeaderImage(sectionAfter, i + 1)
    const src = sectionAfterisImage && sectionAfter.images[0].url

    return (
      <ArtistHeaderSection key={i}>
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg src={src} />
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    return (
      <TextSection size="5" key={i}>
        <div dangerouslySetInnerHTML={{ __html: section.body }} />
      </TextSection>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper key={i}>
        <ImageSetPreview section={section} />
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props

    const renderedSections = sections.map((section, i) => {
      const isHeader = this.sectionIsHeader(section)
      const isHeaderImage = this.sectionIsHeaderImage(section, i)

      if (isHeader) {
        return this.sectionHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text") {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    return (
      <React.Fragment>
        <Nav canFix />

        <ArticleWrapper>
          <Eoy2018ArticleHeader />

          <ArticleContent>{this.getSections()}</ArticleContent>
        </ArticleWrapper>
      </React.Fragment>
    )
  }
}

const ArticleWrapper = styled.div`
  padding: 0 40px;
`

const ArticleContent = styled.div`
  padding: 40px 0;
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
  }
`

const ArtistHeaderSection = styled.div`
  height: 55vh;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
  display: flex;
  margin-bottom: 40px;
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    padding: 20px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    display: inline-flex;
    padding: 20px;
    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }
    &:hover {
      background: ${color("purple100")};
      color: white;
    }
  }
`

const ArtistHeaderImg = styled.div<{ src?: string }>`
  flex: 1;
  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};
`

const TextSection = styled(Serif)`
  max-width: 75%;
  margin-left: auto;
  padding-bottom: 60px;

  p {
    font-size: 24;
  }
`
const ImageSetWrapper = styled.div`
  margin-bottom: 60px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
`
