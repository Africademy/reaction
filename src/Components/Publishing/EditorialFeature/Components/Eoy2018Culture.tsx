import { Box, Flex, Sans } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import { Author, StyledAuthor } from "Components/Publishing/Byline/Author"
import { Date, DateContainer } from "Components/Publishing/Byline/Date"
import { Share } from "Components/Publishing/Byline/Share"
import { getArticleFullHref } from "Components/Publishing/Constants"
import {
  ArticleProps,
  FeatureLayout,
} from "Components/Publishing/Layouts/FeatureLayout"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import React from "react"
import styled from "styled-components"

export const Eoy2018Culture: React.SFC<ArticleProps> = props => {
  const { article } = props
  return (
    <ArticleWrapper>
      <Nav canFix color="black" backgroundColor="white" />

      <ArticleHeader>
        <ArticleTitle size="16" weight="medium">
          <div>Year In Visual Culture</div>
          <div>2018</div>
        </ArticleTitle>
        <HeaderImg src={article.hero_section.url} />
      </ArticleHeader>

      <Intro>
        <Box width={"20%"}>
          <Share
            url={getArticleFullHref(article.slug)}
            title={article.social_title || article.thumbnail_title}
          />
        </Box>
        <Flex>
          <Author authors={article.authors} />
          <Date date={article.published_at} />
        </Flex>
      </Intro>

      <Box>
        <div dangerouslySetInnerHTML={{ __html: article.sections[0].body }} />
      </Box>
      <FeatureLayout {...props} />
    </ArticleWrapper>
  )
}

const ArticleWrapper = styled.div`
  ${NavContainer} {
    border-bottom-width: 6px;
    border-top: 6px solid black;
  }

  blockquote {
    ${unica("s34")};
  }
`

const ArticleTitle = styled(Sans)`
  text-transform: uppercase;
  padding: 20px 40px;
  border-bottom: 6px solid;
`

const Intro = styled(Flex)`
  align-items: center;
  padding: 20px 40px;

  ${DateContainer} {
    margin-top: 0;
  }
  ${StyledAuthor} {
    margin-top: 0;
  }
`

const HeaderImg = styled.img`
  height: 100%;
  object-fit: cover;
  object-position: center;
  width: 100%;
`

const ArticleHeader = styled.div`
  height: calc(100vh - 46px);
  overflow: hidden;
  border-bottom: 6px solid black;
`
