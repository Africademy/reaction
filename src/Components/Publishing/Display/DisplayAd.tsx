import { Box, color, Flex, FlexProps, Sans } from "@artsy/palette"
import * as AnalyticsSchema from "Artsy/Analytics/Schema"
import { useTracking } from "Artsy/Analytics/useTracking"
import { is300x50AdUnit } from "Components/Publishing/Display/DisplayTargeting"
import { AdDimension, AdUnit } from "Components/Publishing/Typings"
import React, { SFC, useState } from "react"
import { Bling as GPT } from "react-gpt"
import styled from "styled-components"
import { useDidMount } from "Utils/Hooks/useDidMount"

export interface DisplayAdProps extends FlexProps {
  adUnit: AdUnit
  adDimension: AdDimension
  targetingData: {
    is_testing: boolean
    page_type: string
    post_id: string
    tags: string
  }
  isSeries?: boolean
  isStandard?: boolean
  articleSlug: string
}

export interface DisplayAdContainerProps extends FlexProps {
  isSeries?: boolean
  isStandard?: boolean
  adDimension?: AdDimension
  isAdEmpty?: boolean
}

GPT.syncCorrelator(true)

export const DisplayAd: SFC<DisplayAdProps> = props => {
  const {
    adDimension,
    adUnit,
    targetingData,
    articleSlug,
    ...otherProps
  } = props
  const [width, height] = adDimension.split("x").map(a => parseInt(a))
  const [isAdEmpty, setAdEmpty] = useState(null)
  const isMobileLeaderboardAd = is300x50AdUnit(adDimension)
  const { trackEvent } = useTracking()

  const onClickAd = () => {
    trackEvent({
      context_page: AnalyticsSchema.PageName.ArticlePage,
      context_module: AnalyticsSchema.ContextModule.AdServer,
      context_page_owner_id: props.targetingData.post_id,
      context_page_owner_slug: props.articleSlug,
      context_page_owner_type: AnalyticsSchema.OwnerType.Article,
    })
  }

  const ad = (
    <GPT
      collapseEmptyDiv
      adUnitPath={`/21805539690/${adUnit}`}
      targeting={targetingData}
      slotSize={[width, height]}
      onSlotRenderEnded={event => {
        setAdEmpty(event.isEmpty)
      }}
    />
  )

  if (isAdEmpty) {
    return null
  }

  if (useDidMount) {
    trackEvent({
      context_page: AnalyticsSchema.PageName.ArticlePage,
      context_module: AnalyticsSchema.ContextModule.AdServer,
      context_page_owner_id: props.targetingData.post_id,
      context_page_owner_slug: props.articleSlug,
      context_page_owner_type: AnalyticsSchema.OwnerType.Article,
    })
  }

  return (
    <DisplayAdContainer
      onClick={onClickAd}
      flexDirection="column"
      pt={isMobileLeaderboardAd ? 2 : 4}
      pb={isMobileLeaderboardAd ? 2 : 1}
      height={
        isAdEmpty || isAdEmpty === null
          ? "1px" // on initial render OR when no ad content returned from Google, set 1px height to ad container to prevent jarring UX effect
          : isMobileLeaderboardAd
          ? "100px" // on mobile 300x50 ads reduce ad container height to 100px
          : "334px"
      }
      isAdEmpty={isAdEmpty}
      {...otherProps}
    >
      <Box m="auto">
        {ad}
        <Sans size="1" color="black30" m={1}>
          Advertisement
        </Sans>
      </Box>
    </DisplayAdContainer>
  )
}

const DisplayAdContainer = styled(Flex)<DisplayAdContainerProps>`
  margin: ${p => (p.isStandard ? "0" : "0 auto")};
  border-top: ${p => (p.isSeries ? `1px solid ${color("black10")}` : "none")};
  background: ${p => (p.isSeries ? color("black100") : color("black5"))};
  text-align: center;
  width: 100%;
  visibility: ${p =>
    p.isAdEmpty || p.isAdEmpty === null ? "hidden" : "visible"};
`
