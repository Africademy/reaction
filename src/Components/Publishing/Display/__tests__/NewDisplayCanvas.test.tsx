import { NewDisplayCanvas } from "Components/Publishing/Display/NewDisplayCanvas"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"

import { StandardArticleHostedAdCanvas } from "../../Fixtures/Components"

jest.mock("sharify", () => ({
  data: {
    HASHTAG_LAB_ADS_ENABLED: true,
    HASHTAG_LAB_ADS_ALLOWLIST: "alloweduser@email.com,alloweduser2@email.com",
    CURRENT_USER: {
      type: "Admin",
      email: "alloweduser@email.com",
    },
  },
}))

describe("snapshot", () => {
  it("renders the new canvas in standard layout", () => {
    const component = renderer
      .create(
        <NewDisplayCanvas
          adDimension={StandardArticleHostedAdCanvas.adDimension}
          adUnit={StandardArticleHostedAdCanvas.adUnit}
        />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})

describe("data", () => {
  it("renders the component with the correct data and properties in standard layout articles", () => {
    const canvas = mount(
      <NewDisplayCanvas
        adDimension={StandardArticleHostedAdCanvas.adDimension}
        adUnit={StandardArticleHostedAdCanvas.adUnit}
      />
    )

    expect(canvas.props().adDimension).toEqual("970x250")
    expect(canvas.props().adUnit).toEqual("Desktop_TopLeaderboard")
    expect(canvas).toHaveLength(1)
    canvas.find({ className: "htl-ad" })
    canvas.find({ "data-sizes": "970x250" })
    canvas.find({ "data-eager": true })
    canvas.find({ "data-unit": "Desktop_TopLeaderboard" })
  })
})
