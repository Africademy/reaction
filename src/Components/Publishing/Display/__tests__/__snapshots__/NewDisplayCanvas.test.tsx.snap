// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders the new canvas in standard layout 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 1250px;
  margin: 0 auto;
  width: 100%;
  min-height: -webkit-fit-content;
  min-height: -moz-fit-content;
  min-height: fit-content;
  max-width: 1250px;
  box-sizing: border-box;
}

<div
  className="c0"
  width="100%"
>
  <div
    className="htl-ad"
    data-eager={true}
    data-sizes="970x250"
    data-unit="Desktop_TopLeaderboard"
  />
</div>
`;
